tsdb/engine/tsm1/file_store.go

```diff
@@ -309,7 +311,7 @@ func NewFileStore(dir string, options ...tsmReaderOption) *FileStore {
 		obs:           noFileStoreObserver{},
 		parseFileName: DefaultParseFileName,
 		copyFiles:     runtime.GOOS == "windows",
-		readerOptions: options,
+		readerOptions: append(options, WithParseFileNameFunc(DefaultParseFileName)),
```

Comment:
To get the expected behavior of `NewFileStore("dirname", tsm1.WithParseFileNameFunc(myParseFunc))`, the `DefaultParseFileName` must be prepended to the options. The appended `DefaultParseFileName` will cause `DefaultParseFileName` to be used instead of `myParseFunc`.